core/spec/models/spree/stock/estimator_spec.rb

```diff
@@ -140,6 +140,32 @@ module Stock
             expect(shipping_rates.first.tax_rate).to eq(tax_rate)
           end
         end
+
+        it 'uses the configured shipping rate selector' do
+          class Spree::Stock::TestSelector; end;
```

Comment:
I went with doubles in this case since all that matters is the selector and sorter implement `find_default` and `sort`. If the implementation changes to use a different method or renames them, then you would need to update the test in either case.

I'm curious as to the reasoning behind implementing the method in the test class.

Also, it appears the global state gets cleared :)